discord/userDataSet.go

```diff
@@ -47,6 +47,17 @@ func (uds *UserDataSet) AddFullUser(user game.UserData) {
 	uds.lock.Unlock()
 }
 
+func (uds *UserDataSet) containsUser(userId string) bool{
```

Comment:
I think this could be made more compact/lock-efficient using:
```
uds.lock.RLock() //no need for a full write lock, read lock is fine
defer uds.lock.RUnlock()

_, ok := uds.userDataSet[userId]
return ok
```

Reply:
Also maybe "userID" instead of "userId"

Reply:
Thanks for the information! I have implemented your changes!